### PR TITLE
Feat/v3 cache warmer and other tech debt cleanup

### PR DESCRIPTION
**Pull request by khcoaching**
Cleans up the following tech debt:
- use `Box` containers instead of div flex
- configure root level `lint:fix` with concurrency
- Fix pool searching to support spaces
- Upgrade Husky commit hooks to latest (`yarn && yarn prepare`)

Updates the memoizer to support force updating the cache. Adds a new worker to keep cache warm. It will fetch topPools and top 10 pairs and force a cache update at every interval.

If the cache warmer goes down for whatever reason, no worries, the og function is still memoized

### CODE REVIEW

**Comment by khcoaching** [comment]
@kkennis @luckyOdisetti ready for review

@zmanian for vis

File: packages/server/src/util/memoizer-redis/index.ts [hunk 1]
```diff
@@ -90,7 +92,10 @@ export default function memoizerFactory(client: Redis.Redis, opts: Partial<Memoi
 
     // the actual memoized function
     memoized = async function memoized(...args: Array<any>): Promise<T> {
-      const cacheKey = getCacheKey(keyPrefix, fnKey, hashArgs(args));
+      let cacheKey = getCacheKey(keyPrefix, fnKey, hashArgs(args));
+      if (typeof cacheKeyOverride === 'function') {
+        cacheKey = cacheKeyOverride(cacheKey, keyPrefix, fnKey, args);
```
Comment (kkennis) [inline packages/server/src/util/memoizer-redis/index.ts]: In what scenario would the override come into play?

File: packages/workers/src/scheduler.ts [hunk 2]
```diff
@@ -15,6 +16,11 @@ cron.schedule(CRON_EVERY_HOUR, () => {
     void runDiscordAlerts();
 });
 
+// every 5 minutes
+cron.schedule('5/* * * * *', () => {
+    void runMainnetCacheWarmer();
```
Comment (kkennis) [inline packages/workers/src/scheduler.ts]: Great idea to move this outside the server code

**Review by kkennis — DISMISSED** [review]
Good stuff 👍 

Looks like the build failure is legit, but it's only in a spec file

File: packages/server/src/util/memoizer-redis/index.ts (hunk 1 above)
Comment (khcoaching) [reply packages/server/src/util/memoizer-redis/index.ts]: i dreamt about this 😂 . this idea was only valid for moving to the workers but then I found out you already exported the v2 fetcher. I'm going to just use the same fn.

File: packages/server/src/util/memoizer-redis/index.ts [hunk 3]
```diff
@@ -122,6 +122,31 @@ export default function memoizerFactory(client: Redis.Redis, opts: Partial<Memoi
       return finalResult;
     }
 
+    // Force a cache update
```
Comment (khcoaching) [inline packages/server/src/util/memoizer-redis/index.ts:125]: @kkennis way cleaner interface, dunno what i was thinking last night

File: packages/workers/src/scripts/redis-cache-warmer.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+const count = 1000;
+const sort = 'volumeUSD';
+
+export default async function run () {
+  // this function already validates null and length 0 pools
+  // it will also throw on error, so only valid data will be cached
+  console.log('Fetching Top Pools and updating cache');
+  console.time('update-top-pools-elapsed');
+
+  const topPools = await getTopPools.forceUpdate(count, sort);
```
Comment (khcoaching) [inline packages/workers/src/scripts/redis-cache-warmer.ts]: @kkennis makes this part way cleaner too

File: packages/workers/src/scripts/redis-cache-warmer.ts [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+  const topTen = topPools.slice(0, 10);
+  const topTenSymbols = topTen.map(poolName);
+
+  console.log(`Updating pool data for top ten pools ${topTenSymbols}`);
+  for (const pool of topTen) {
+    console.time(`update-pool-${poolName(pool)}-${pool.id}-elapsed`);
+
+    // must pass undefined to match signature of original call
+    await getPoolOverview.forceUpdate(pool.id, undefined);
```
Comment (khcoaching) [inline packages/workers/src/scripts/redis-cache-warmer.ts]: now we can do this

File: packages/server/src/api/util/catch-async-route.ts [hunk 6]
```diff
@@ -13,8 +16,7 @@ export default function catchAsyncRoute(
             const result = await controllerFn(req, res);
             return res.json(result);
         } catch (err) {
-            console.log('Error in Route:', err);
-            res.status(err.status).send(err);
+            next(err);
```
Comment (khcoaching) [inline packages/server/src/api/util/catch-async-route.ts:19]: @kkennis looks like this change made it back in some how and i had to revert back to using `next(err)`

File: packages/server/test/api/controllers/pool.spec.ts [hunk 7]
```diff
@@ -14,420 +21,445 @@ const validId = '0x1581d1a4f79885255e1993765ddee80c5e715181';
...
-    jest.restoreAllMocks();
-  });
-
-  describe('GET /ethprice', () => {
-    const url = '/api/v1/mainnet/ethprice';
-    let getEthPrice;
-    beforeEach(() => {
-      getEthPrice = jest.spyOn(fetcher, 'getEthPrice');
+    afterEach(() => {
+        jest.restoreAllMocks();
```
Comment (khcoaching) [inline packages/server/test/api/controllers/pool.spec.ts:25]: looks like prettier went to town on this file

File: packages/server/tsconfig.build.json [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+        "rootDir": "src",
+        "baseUrl": "./src",
+        "plugins": [
+            {
+                "transform": "@zerollup/ts-transform-paths"
+            }
+        ]
+    },
+    "include": [
+        "src"
```
Comment (khcoaching) [inline packages/server/tsconfig.build.json:15]: build tsconfig does not include test

File: packages/server/tsconfig.json [hunk 9]
```diff
@@ -12,11 +12,11 @@
         ]
     },
     "include": [
-        "src"
+        "src",
+        "test"
```
Comment (khcoaching) [inline packages/server/tsconfig.json:16]: server root tsconfig must include test

**Comment by khcoaching** [comment]
going to close this and reopen

**Pull request closed without merging**